### PR TITLE
[Mini Challenge 3] Hooks

### PR DESCRIPTION
**Pull request by saulmeneses**
- Updated Header component (search bar & dark mode switch).
- Added global style rule.
- Generated custom hook for youtube videos fetch task (useYoutubeSearch).
- Added test cases for custom hook.
- Updated home page.
- Added video details page.
- Applied tests on most components; sharing coverage results (test-coverage.txt, test-capture.png).

### CODE REVIEW

File: src/components/ContentCard/ContentCard.component.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  return (
+    <StyledCard onClick={handleOnClick}>
+      <CardActionArea>
+        <StyledCardMedia
+          src={item.snippet.thumbnails.high.url}
+          title={item.snippet.title}
+        />
+        <CardContent>
+          <Title dangerouslySetInnerHTML={{ __html: item.snippet.title }} />
```
Comment (jparciga) [inline src/components/ContentCard/ContentCard.component.jsx]: Is this `dangerouslySetInnerHTML` really necessary? I think you could do something like this instead:

```suggestion
          <Title>{ item.snippet.title}</Title>
```

File: src/components/ContentDetails/__tests__/ContentDetails.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      <Router history={historyMock}>
+        <ContentDetails item={mockItems[0]} relatedItems={mockItems} />
+      </Router>
+    );
+
+    const element = screen.getByLabelText('back button');
+
+    fireEvent.click(element);
+    expect(historyMock.push).toHaveBeenCalled();
+  });
```
Comment (jparciga) [inline src/components/ContentDetails/__tests__/ContentDetails.spec.jsx:94]: This is a good test case 👍🏻

**Review by jparciga — COMMENTED** [review]
# Mini-Challenge 3 Feedback:
Good job! It's great that you also have implemented Routing for this challenge.

## Acceptance Criteria:
✅  Videos in the Home View are fetched from the YouTube API and displayed correctly.
✅  Users can search for YouTube videos using the search field on the header.
✅  A video can be played from the Video Details View after clicking on it.
✅  The video information and related videos list are displayed correctly on the Video Details View.
✅  When a user clicks on a related video the video player, information, and related videos list are updated.

## Bonus Points
Custom Hooks for API calls are implemented and tested correctly.
✅  Testing coverage is above 60%. (Please include a screenshot of the code coverage report in your PR description).

File: src/components/ContentCard/ContentCard.component.jsx (hunk 1 above)
Comment (saulmeneses) [reply src/components/ContentCard/ContentCard.component.jsx]: I mostly use it for the elements to interpret HTML codes from string values. If there is another way, please, I'm open for advices.

**Pull request closed without merging**